Gems/RobotecSplineTools/Code/Source/Clients/RobotecSplineToolsSystemComponent.cpp

```diff
@@ -0,0 +1,77 @@
+
+#include "RobotecSplineToolsSystemComponent.h"
+
+#include <RobotecSplineTools/RobotecSplineToolsTypeIds.h>
+
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace RobotecSplineTools
+{
+    AZ_COMPONENT_IMPL(RobotecSplineToolsSystemComponent, "RobotecSplineToolsSystemComponent",
+        RobotecSplineToolsSystemComponentTypeId);
+
+    void RobotecSplineToolsSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<RobotecSplineToolsSystemComponent, AZ::Component>()
+                ->Version(0)
+                ;
+        }
+    }
+
+    void RobotecSplineToolsSystemComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("RobotecSplineToolsService"));
+    }
+
+    void RobotecSplineToolsSystemComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("RobotecSplineToolsService"));
+    }
+
+    void RobotecSplineToolsSystemComponent::GetRequiredServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+    }
+
+    void RobotecSplineToolsSystemComponent::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+    }
```

Comment:
no need to implement those methods if the dependencies are not updated

Reply:
Theoretically, yes, but those are called by EditorSystem component. I can remove this boiler plate code, but probably we will add some Game components (e.g. better visualization for splines in the game). So I would rather keep some boilerplate here for future use.